pkgs/development/libraries/science/math/or-tools/default.nix

```diff
@@ -41,30 +41,38 @@ stdenv.mkDerivation rec {
     })
   ];
 
+  # or-tools normally attempts to build Protobuf for the build platform when
+  # cross-compiling. Instead, just tell it where to find protoc.
+  postPatch = ''
+    echo "set(PROTOC_PRG $(type -p protoc))" > cmake/host.cmake
```

Comment:
type -p silently fails if protoc is not available. We should either take it from PATH directly or hardcode the path here.

Reply:
Hardcoding the path is problematic because we need the version of protoc (from build platform protobuf) to match the version of host platform protobuf. On 32-bit platforms, an older version of protobuf is currently used, and `python3Packages.protobuf` uses `propagatedNativeBuildInputs` to get the matching version of protoc in `PATH`, but if we used `buildPackages.protobuf`, we would get a newer version.

If we just do `set(PROTOC_PRG protoc)`, then CMake interprets it as a target rather than a path.

Lastly, if `type -p` fails, `PROTO_PRG` will be empty and the build will fail, so it won't really be a silent failure.